src/Parse.idr

```diff
@@ -38,56 +41,101 @@ fromVMax {m} vm = go lteRefl vm
 
 ---
 
+-- TODO add to TParsec
+
+MonadTrans (ResultT e) where
+  lift = MkRT . map Value
+
+MonadTrans (TParsecT e a) where
+  lift = MkTPT . lift . lift
+
+andbindm : Monad mn =>
+  All (Parser mn p a :-> Cst (a -> mn b) :-> Parser mn p (a, b))
+andbindm p f = MkParser $ \mlen, ts => do ra <- runParser p mlen ts
+                                          b <- f (Value ra)
+                                          pure $ map (flip MkPair b) ra
+
+landbindm : Monad mn =>
+  All (Parser mn p a :-> Cst (a -> mn b) :-> Parser mn p a)
+landbindm p f = map fst (andbindm p f)
```

Comment:
Is there ever an advantage to use `landbindm p f` instead of just `p`? In case you want to allow `f` the possibility to fail, or cause a side-effect?

Reply:
Yeah, I added it just for the sake of completeness, with side effects in mind. I'm going to move this section to `TParsec` proper after this gets merged, so maybe Guillaume will have to say something about it :)